### DISCUSSION TITLE
is there OTA support?

### DISCUSSION THREAD

**Discussion by yvolchkov** [luar123/zigbee_esphome#14] [Q&A]
it turns out wifi is not working if Zigbee is on on esp32-c6. For me this ruins benefits of Esphome as I can't flash device over the air . Perhaps it is possible to switch off Zigbee on demand so device gets online in the wifi network and can be refreshed? Or maybe there is a way to use ota over zigbee with zigbee2mqtt? However the second is painfully slow, I would prefer switching back to wifi for the time of updating

**Comment by luar123** [comment]
Zigbee OTA would be cool, but is not planned at the moment.
Wifi coexistence should work since PR from @JochenFriedrich is merged: https://github.com/luar123/zigbee_esphome/commit/c806486fa9dd5176cdfe0fc443c8fd538f4403b3
Please let me know if OTA is working with this.

**Reply by JochenFriedrich** [reply]
That was exactly my motivation for this patch. esp_coex_wifi_i154_enable() does the trick.

Zigbee OTA wouldn't be easy at least for z2m as you would definitely need a custom converter with ota manually added to the automatic generated one. You also would need to track an 32bit upgrade_file_version which needs to be greater than the last uploaded one. And as you said it would take hours for a single update.

**Reply by fricy** [reply]
@JochenFriedrich 
May I ask what's the reasoning behind preventing AP mode?

https://github.com/luar123/zigbee_esphome/blob/dd7fdaf83cb6b5f23b61f4107f8e656ebd19fe41/components/zigbee/__init__.py#L231

**Reply by luar123** [reply]
See https://docs.espressif.com/projects/esp-idf/en/stable/esp32c6/api-guides/coexist.html
Zigbee router + AP is not supported at all, end device + AP is marked unstable. Even Router+station is marked unstable but I guess it is ok when used only for OTA.

The main issue is that in AP mode and in zigbee router mode the esp32 needs to have RX enabled all the time, but it cannot listen on both networks simultaneously.

**Reply by fricy** [reply]
But if I don't use them simultaneously it should be ok, correct? Turn off Zigbee, fire up the AP for admin tasks, etc.

**Reply by luar123** [reply]
As long as you use `router: false` it should work. Unfortunately there is no way to turn off zigbee, but WiFi should be able to handle some package loss.

**Comment by Hedda** [comment]
> maybe there is a way to use ota over zigbee with zigbee2mqtt?

For reference, started a new feature request discussion asking for native Zigbee OTA Upgrade Cluster update support in upstream ESPHome:

* https://github.com/orgs/esphome/discussions/3573

Note that Espressif's upstream ESP Zigbee SDK does at least have support for Zigbee OTA Upgrade Cluster related APIs so hopefully someone else might step-up and implement this if it is made clear that no one else is planning this. See:

* https://docs.espressif.com/projects/esp-zigbee-sdk/en/latest/esp32/user-guide/zcl_ota_upgrade.html

and

  * https://docs.espressif.com/projects/esp-zigbee-sdk/en/latest/esp32/api-reference/esp_zigbee_ota.html

Requiring dual-protocol support to update via WiFi, Bluetooth, or Serial OTW will not work for all use cases and is not really user-friendly, so while those can be used as workarounds in the short term or optinal update solutions in the future I hope that in the long term there will eventually be a time when ESPHome support native Zigbee OTAU so can have the Zigbee gateway provide the updates via Zigbee OTA, (so there will be no need to mess around with WiFi, Bluetooth, or Serial OTW when ESPHome been configured to be a Zigbee device).

Again these are just my personal opinions however I believe that sooner or later ESPHome will also need to have native support for building firmware update images for native Zigbee OTA (Over-The-Air) updates, however I too guess that it would probably be a good idea to enable using Bluetooth for OTA firmware updates as a workaround until then if that is simpler to add support for in ESPHome than native architecture for Zigbee OTAU Firmware (ZigBee Over-the-Air Upgrade) inside ESPHome. Anyway, I imagine that if and when native Zigbee OTA support is added to ESPHome then it will create a new opportunity to make and sell finished Zigbee devices based on ESPHome that are not only meant for DIY enthusiasts but also for less advanced users who are just willing to buy an ESPHome device if it 100% works as a native Zigbee device, including supporting Zigbee OTA updates,

PS: The README should also mention that this is currently not planned:

* https://github.com/luar123/zigbee_esphome/pull/89
